barretenberg/cpp/src/barretenberg/stdlib/encryption/ecdsa/ecdsa_impl.hpp

```diff
@@ -18,7 +18,63 @@ auto& engine = numeric::get_debug_randomness();
 }
 
 /**
- * @brief Verify ECDSA signature. Produces unsatisfiable constraints if signature fails
+ * @brief Validate the inputs used by the verification function and return messages if they produce an invalid circuit.
+ *
+ * @tparam Builder
+ * @tparam Curve
+ * @tparam Fq
+ * @tparam Fr
+ * @tparam G1
+ * @param hashed_message
+ * @param public_key
+ * @param sig
+ * @param scalar_mul_result
+ */
+template <typename Builder, typename Curve, typename Fq, typename Fr, typename G1>
+void validate_inputs(const stdlib::byte_array<Builder>& hashed_message,
+                     const G1& public_key,
+                     const ecdsa_signature<Builder>& sig)
+{
+    auto assert_smaller_than = [](const uint512_t& value,
+                                  const uint256_t& max_value,
+                                  const std::string& value_label,
+                                  const std::string& max_value_label) {
+        std::string msg =
+            "The " + value_label + " is bigger than " + max_value_label + ". This will produce an unsatisfied circuit.";
+        if (value > max_value) {
```

Comment:
`value` is `uint512_t` and `max_value` is `uint256_t`. I think both should be `uint512_t`?